test/snapshots/new-for-builtins.js.md

```diff
@@ -1388,3 +1367,129 @@ Generated by [AVA](https://avajs.dev).
         | 	      ^^^^^^^^^^^^^^^^^^ Use \`new Map()\` instead of \`Map()\`.␊
       3 | }␊
     `
+
+## invalid(1): const foo = Date();
+
+> Input
+
+    `␊
+      1 | const foo = Date();␊
+    `
+
+> Output
+
+    `␊
+      1 | const foo = String(new Date());␊
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | const foo = Date();␊
+        |             ^^^^^^ Use \`String(new Date())\` instead of \`Date()\`.␊
+    `
+
+## invalid(2): const foo = globalThis.Date();
+
+> Input
+
+    `␊
+      1 | const foo = globalThis.Date();␊
+    `
+
+> Output
+
+    `␊
+      1 | const foo = String(new Date());␊
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | const foo = globalThis.Date();␊
+        |             ^^^^^^^^^^^^^^^^^ Use \`String(new Date())\` instead of \`Date()\`.␊
+    `
+
+## invalid(3): function foo() { return(globalThis).Date(); }
+
+> Input
+
+    `␊
+      1 | function foo() {␊
+      2 | 	return(globalThis).Date();␊
+      3 | }␊
+    `
+
+> Output
+
+    `␊
+      1 | function foo() {␊
+      2 | 	return String(new Date());␊
+      3 | }␊
+    `
+
+> Error 1/1
+
+    `␊
+      1 | function foo() {␊
+    > 2 | 	return(globalThis).Date();␊
+        | 	      ^^^^^^^^^^^^^^^^^^^ Use \`String(new Date())\` instead of \`Date()\`.␊
+      3 | }␊
+    `
+
+## invalid(4): const foo = Date(/*comment*/);
+
+> Input
+
+    `␊
+      1 | const foo = Date(/*comment*/);␊
+    `
+
+> Error 1/1
+
+    `␊
+    > 1 | const foo = Date(/*comment*/);␊
```

Comment:
Why no fix this to

```js
const foo = String(new Date(/*comment*/));
```

It should be safe

Reply:
It's suggestion, see L1455

Reply:
Sorry, I misunderstood, we should try not remove comments.

Reply:
The correct fix seems like

```js
		function * fix(fixer) {
			yield * fixSpaceAroundKeyword(fixer, node, sourceCode);
			yield fixer.insertTextBefore(node, 'String(');
			yield fixer.insertTextAfter(node, ')');
			yield * switchCallExpressionToNewExpression(node, sourceCode, fixer, /* fixSpaces */ false);
			yield * removeArguments();
		}
```

Just not worth the effort

Reply:
Maybe you wonder why `switchCallExpressionToNewExpression` is necessary, something like

```js
Array(globalThis)[0].Date(/*comment*/);
````

is not easy to fix correctly, but `switchCallExpressionToNewExpression` already considered this.